pkg/event/sns.go

```diff
@@ -0,0 +1,59 @@
+package event
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sns/snsiface"
+)
+
+// SnsEvent is for publishing events to SQS
+type SnsEvent struct {
+	sns      snsiface.SNSAPI
+	topicArn arn.ARN
+}
+
+// Publish an event to the topic
+func (s *SnsEvent) Publish(i interface{}) error {
+	bodyJSON, err := json.Marshal(i)
+	if err != nil {
+		return errors.New("Unable to marshal response")
+	}
+
+	// Wrap the body in a SNS message object
+	message, err := json.Marshal(struct {
```

Comment:
this could be less verbose, just as:

```
message, err := json.Marshal(map[string]string{
		"default": string(bodyJSON),
		"Body": string(bodyJSON)
	})

Reply:
Thanks updated.